### PR TITLE
Removed datetime_cast_sql, which is never overridden or used anywhere in Django.

### PR DESCRIPTION
**Pull request by shaib**
Done as a PR rather than a direct commit to give some chance to interested parties to stop it

### CODE REVIEW

**Comment by mjtamlyn** [comment]
Ironically, this may actually be useful for #4778

**Comment by timgraham** [comment]
Marc, since the PR you referenced has been closed, can we move forward with this?

File: docs/releases/1.9.txt [hunk 1]
```diff
@@ -359,6 +359,11 @@ Database backend API
 * To use the new ``date`` lookup, third-party database backends may need to
   implement the ``DatabaseOperations.datetime_cast_date_sql()`` method.
 
+* The ``DatabaseOperations.datetime_cast_sql()`` (without "``_date_``") method
```
Comment (timgraham) [inline docs/releases/1.9.txt]: I'd drop the parenthetical note as it just confused me.

File: docs/releases/1.9.txt [hunk 2]
```diff
@@ -359,6 +359,11 @@ Database backend API
...
+  has been removed. This method served to format dates on Oracle long
+  before 1.0, but hasn't been overridden by any core backend in years,
```
Comment (timgraham) [inline docs/releases/1.9.txt]: drop the last comma since "hasn't been..." can't be its own sentence.

**Comment by mjtamlyn** [comment]
Yup I'm fine for this to be moved on so long as it still applies.

**Comment by aaugustin** [comment]
Removing dead code :+1:

**Comment by shaib** [comment]
closed via eecd42e

**Pull request closed without merging**